sos/report/plugins/ceph_osd.py

```diff
@@ -39,9 +39,60 @@ def setup(self):
 
         self.add_cmd_output([
             "ceph-disk list",
-            "ceph-volume lvm list",
+            "ceph-volume lvm list"
         ])
 
+        ceph_cmds = [
+           "bluestore bluefs available",
+           "config diff",
+           "config show",
+           "dump_blacklist",
+           "dump_blocked_ops",
+           "dump_historic_ops_by_duration",
+           "dump_historic_slow_ops",
+           "dump_mempools",
+           "dump_ops_in_flight",
+           "dump_op_pq_state",
+           "dump_osd_network",
+           "dump_reservations",
+           "dump_watchers",
+           "log dump",
+           "perf dump",
+           "perf histogram dump",
+           "objecter_requests",
+           "ops",
+           "status",
+           "version",
+        ]
+
+        osd_ids = []
+        out = self.exec_cmd('ps -u ceph -o args')
+
+        if out['status'] == 0:
+            # Get the running OSD process IDs from ps
+            osd_ids = [
+                'osd.%s' % m.split()[5] for m in out['output'].splitlines()
+                if 'ceph --id' in m and len(m.split()) > 5
+                and m.split()[5].isdigit() and m.split()[4] == "--id"
+            ]
```

Comment:
We are redundantly calling `m.split()` many times in one iteration of `for m in ..`. But calling it once would mean less obvious/elegant code and I hope/expect the data are kept in some python's/kernel cache, and the "`ps`" output isnt too big.

TL;DR: I am not happy with this, but I see it as best option.

Reply:
If we're going to expand the safeguard checks this much, we don't _need_ to keep everything in a single list constructor. The request to do this originally was because there was only a single check occurring and there were 2 lists being used to create the single list of osd ids.

```python
osd_ids = []
for procs in out['output'].splitlines():
    proc = procs.split()
    if len(proc) < 6:
        continue
    if proc[4] == '--id' and proc[5].isdigit():
        osd_ids.append("osd.%s" % proc[5])
```

Single `split()` call, and is very easily grok'd.

Reply:
+1 to this.

Reply:
leaving as is, thank you